org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/corrections/proposals/UnInitializedFinalFieldSubProcessor.java

```diff
@@ -0,0 +1,92 @@
+/*******************************************************************************
+ * Copyright (c) 2019 IBM Corporation and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copied from /org.eclipse.jdt.ui/src/org/eclipse/jdt/internal/ui/text/correction/UnInitializedFinalFieldSubProcessor.java
```

Comment:
Since copying the duplicated code to jdt.ls is not the best practice, would you consider moving it to jdt.core.manipulation for jdt.ls reuse?

Reply:
This could be possible. `UnInitializedFinalFieldSubProcessor` doesn't seem to require many classes from jdt.ui. The main one is `InitializeFinalFieldProposal`.

- Usage of IInvocationContext, might be replaced with just ICompilationUnit (of IJavaElement) & CompilationUnit (of ASTNode) as that's what is mainly needed.
- IProblemLocation -> IProblemLocationCore
- We can probably use something other than ICommandAccess to make this work. Maybe even an Object collection in core if there's nothing in common.

Reply:
Looking back, this is a little more complicated than I thought.

`UnInitializedFinalFieldSubProcessor` uses `InitializeFinalFieldProposal` which uses `LinkedCorrectionProposal` which we have copied over. It might still be possible as I'm not seeing any huge deal breakers, but there may be quite a few things to refactor in JDT.UI before we can eliminate the copied code.

Reply:
@JessicaJHee , can you have a look at migrating this and see if it's possible ? Above, I said it can't be done because of `LinkedCorrectionProposal` but @datho7561  did a very similar migration in https://github.com/eclipse-jdt/eclipse.jdt.ui/commit/c9c7e9d3242e1211ac2fc25b91fb22f65203314a#diff-bdce58a19b4250ecb262ca0a1f74742bfdd56a253cbd4ce3c10120939175e433L47 . If you could create a PR similarly to that we can unblock and get this in.

Reply:
My newest PRs for jdt create UnInitializedFinalFieldBaseSubProcessor as a superclass.